### PR TITLE
Plane:force min transition timer to 2 sec

### PR DESCRIPTION
**Pull request by Hwurzburg**
tested in SITL

### CODE REVIEW

**Comment by Rolf-G** [comment]
Why do you want to force everyone by hardcoding to let the min_transition last at least 2 seconds ? 
I would be interested in what this is good for and why of all things 2 seconds ?

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -1699,14 +1699,15 @@ void SLT_Transition::update()
         // after airspeed is reached we degrade throttle over the
         // transition time, but continue to stabilize
         const uint32_t transition_timer_ms = now - transition_low_airspeed_ms;
-        if (transition_timer_ms > (unsigned)quadplane.transition_time_ms) {
+        float trans_time_ms = constrain_float(quadplane.transition_time_ms,2000,20000);
```
Comment (IamPete1) [inline ArduPlane/quadplane.cpp:1702]: I don't think we should constrain on the upper side. You might as well take the opportunity to const it too.
```suggestion
        const float trans_time_ms = MAX(quadplane.transition_time_ms, 2000);
```

Comment (Hwurzburg) [reply ArduPlane/quadplane.cpp:1702]: will add const....upper limit was Tridges request

**Comment by Hwurzburg** [comment]
@Rolf-G because if the EKF has a bad wind estimate, the synthetic airspeed source (no airspeed sensor being used) will switch from EKF3 to DCM when switching into assist... You cant see it in the log but CTUN.As is oscillating rapidly due to this...giving the vehicle some time to accelerate during the airspeed wait period will help reduce this issue by helping get the EKF wind estimator a little time to get happy...wont 100% fix but make it less likely to happen...the transition time is there for a reason, to slow the transition from vtol to fw impact on the EKF and airspeed estimates....the filtering of the fly forward state is being looked at also

File: ArduPlane/quadplane.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/quadplane.cpp:1702]: Do we have a example of a tool long time causing a issue?

A arming check would make me happier, that makes it very clear that we consider the old value invalid and there will be a change in behavior. I much prefer the "why won't my vehicle arm" issues to the "may vehicle just did something different to what it did before and then crashed" issues.

Comment (Hwurzburg) [reply ArduPlane/quadplane.cpp:1702]: too long is a battery consumption issue at least

Comment (Hwurzburg) [reply ArduPlane/quadplane.cpp:1702]: currently its 30sec in the param, maybe just match that?

**Comment by Rolf-G** [comment]
@Hwurzburg It is possible that the problem lies elsewhere. Today I could reproduce the oscillation (here the log: https://magentacloud.de/s/RXMG7ykFD4rgFeR  ) without CTUN.As oscillating. 

But also otherwise I find the procedure of imposing a post-transition time of 2 seconds also on airplanes with airspeed sensor due ti hardcoding not very logical nor user-friendly.
Why is it not enough to set the lower limit in the parameter options ? 
2 seconds are a long time for airplanes with higher transition speed and airspeedsensor.

**Comment by IamPete1** [comment]
In the case of tiltrotors we could wait until the time is up or the motors have reached there fully forward tilt. On this vehile it takes a little less than half a second for the tilts to slew. 

image

**Comment by magicrub** [comment]
I'm not understanding the reasoning for this. The param clearly says _MS so it would be a gross oversight to put the wrong units in there.

**Comment by Hwurzburg** [comment]
Tridge says that the ekf needs some time to switch from VTOL (while in trans timeout) to Fixedwing (transition done)...abrupt changes, with either airspeed sensor or synth airspeed....will be discussed in a few...

**Comment by tridge** [comment]
there are several reasons for the minimum:
 - there is considerable uncertainty in airspeed sensors, spiking above the target airspeed should not complete the transition
 - the code relies on a ramp down time to mean we are in true fly-fwd (zero sideslip) before we tell the EKF to start assuming fly fwd
 - we also rely on ramp down to not cause a very sudden change in aerodynamics, we need that tapering of motors
 - the battery cost is very low as we taper throttle, costs almost zero battery

the max limit of 30s is because this is a signed int16_t, so current max is 32.6s, this makes it a round number

**Comment by Rolf-G** [comment]
Sorry I can't relate to the EKF takes time as a cause of oscillation. On Discord Tim Tuxworth had also described these oscillations with airspeed turned off , but Q_TRANSITION_MS  5000 ms (!)
https://discord.com/channels/674039678562861068/998765238465867876/1158809059756085368

**Comment by Hwurzburg** [comment]
two pieces of the puzzle...short transition times exacerbate the issue and longer ones have value even with the airspeed sensor running to have a minimum ramp time on vtol throtlle to fw throttle which is what happens in the trans timeout phase...but the oscillation has its underlying cause in how the fly forward assumption is applied in the EKF causes it to switch from its synth airspeed to DCMs airspeed on fast trans timeouts before the ekf can get its estimate fixed in the face of a bad wind estimate...all convoluted and will take the ekf guru's involvement to sort out....

**Comment by timtuxworth** [comment]
Just want to point out that my example was flying on a day with zero windspeed. There is an anemometer at the field where I was flying and it wasn't moving.

**Pull request merged**